packages/core/src/tw/command/handlers/state.ts

```diff
@@ -14,6 +16,49 @@ function moveCursorByModelOffset(serviceRegistry: IServiceRegistry, modelOffset:
     serviceRegistry.getService('state').applyTransformation(tn);
 }
 
+export const alignment: ICommandHandler = async (serviceRegistry, textAlign: IParagraphStyle["textAlign"]) => {
+    const cursorService = serviceRegistry.getService('cursor');
+    const renderService = serviceRegistry.getService('render');
+    const modelService = serviceRegistry.getService('model');
+    const stateService = serviceRegistry.getService('state');
+    if (!cursorService.hasCursor()) {
+        return;
+    }
+    const tn = new Transformation();
+    const { anchor, head } = cursorService.getCursorState();
+    let node = modelService.resolvePosition(
+        renderService.convertOffsetToModelOffset(Math.min(anchor, head))
+    ).getChild()
+    if (!node) {
+        return;
+    }
+    let collapsedAt = renderService.convertOffsetToModelOffset(Math.min(anchor, head)) - node.getOffset()
+    tn.addOperation(new AttributeOperation(collapsedAt, {
+        textAlign,
+    }));
+    let index = renderService.convertOffsetToModelOffset(Math.min(anchor, head))
+    const maxIndex = renderService.convertOffsetToModelOffset(Math.max(anchor, head))
+    console.log({
+        index,
+        maxIndex,
+        anchor,
+        head
+    });
+    
+    while (index < maxIndex) {
+        index += 1;
+        const token = stateService.getTokens()[index];
+        if (identifyTokenType(token) === "OpenToken") {
+            if (identifyTokenModelType(token as IOpenToken) === "Block") {
```

Comment:
Instead of asserting on the model node type, we should check the component ID and part ID to identify the appropriate token to apply the attribute on.

The model node type is meant to enforce document structure. For example, as a rule, a block element cannot be a child of an inline element.

On the other hand, the component ID identifies the component (modules in `tw/component/components`), and part ID identifies a part of a component, as a component may have more than one part. There has not been any use cases for multiple parts per component yet, but I imagine something like a quote may have a body part and an author part.

So for paragraph alignment, you'd want to apply the attribute on the paragraph component and the paragraph part.

Reply:
I don't get your point on this. I think you meant that creating `identifyTokenModelType` is unnecessary, right? Actually I have a reason for that, paragraphs are not the only time that has alignments. In the future, we'll have tables and other block components that have their own alignments. I've created this function to identify all `block` components and not just paragraph